smg-automotive/components-pkg
# feat: entitlement support for navigation links

References https://autoricardo.atlassian.net/browse/VSST-1344

## Motivation and context

We need to display links based on user entitlements, which we don't support at the moment. 
We are adding support for entitlements by extending navigation/link configuration.

## Before

Without entitlements support

## After

Entitlements support enabled.

## How to test

Since any modification to navigation link items will affect production navigation, I will provide separate branch environment for testing.

https://github.com/smg-automotive/components-pkg/pull/727

https://testing-entitlements-support-components-pkg.branch.autoscout24.dev/?path=/story/patterns-navigation-header--professional

Open controls tab and choose entitlements. Based on link item required entitlement user should see different link paths. Also there is option to show `missingEntitlementIcon` in case entitlement is missing since showing additional icon is requirement when user doesn't have required entitlement.

Only first two links inside drawer require entitlements (first - `top-listing`, second - `top-car`) .  Look at the [configuration](https://github.com/smg-automotive/components-pkg/pull/727)
![Screenshot 2023-09-07 at 11 21 45](https://github.com/smg-automotive/components-pkg/assets/12614035/539b6a80-9f83-47ad-b76a-aeb1b6d7dd6e)
